### PR TITLE
Tlwp 467 referral email to employers

### PR DESCRIPTION
**Pull request by mikeymonster**

### CODE REVIEW

File: src/Sfa.Tl.Matching.Application/Services/ReferralService.cs [hunk 1]
```diff
@@ -39,38 +41,90 @@ public class ReferralService : IReferralService
...
+                foreach (var qualificationShortTitle in referral.QualificationShortTitles)
+                {
+                    sb.AppendLine($"* {qualificationShortTitle}");
+                }
+                sb.AppendLine(""); //Need a blank line, otherwise the next heading won't be formatted
+            }
+
+            tokens.Add("providers_list", sb.ToString());
+
+            await _emailService.SendEmail(emailTemplate.TemplateName,
```
Comment (adamesfa) [inline src/Sfa.Tl.Matching.Application/Services/ReferralService.cs]: Might be worth passing in DTO

File: src/Sfa.Tl.Matching.Application/Services/ReferralService.cs [hunk 2]
```diff
@@ -39,38 +41,90 @@ public class ReferralService : IReferralService
...
+                "Industry Placement Matching Referral",
+                tokens,
+                "mike.wild@digital.education.gov.uk");
+
+            await SaveEmailHistory(emailTemplate, tokens, opportunityId, toAddress, employer.CreatedBy);
+        }
+
+        private static string GetNumberOfPlacements(bool? placementsKnown, int? placements)
+        {
+            return placementsKnown.GetValueOrDefault()
```
Comment (adamesfa) [inline src/Sfa.Tl.Matching.Application/Services/ReferralService.cs]: We have the same logic in the UI as this...so might be worth moving to a common place

File: src/Sfa.Tl.Matching.Web/Controllers/OpportunityController.cs [hunk 3]
```diff
@@ -89,6 +89,9 @@ public async Task<IActionResult> CheckAnswersReferrals(CheckAnswersReferralViewM
             var dto = _mapper.Map<CheckAnswersDto>(viewModel);
             await _opportunityService.UpdateOpportunity(dto);
 
+            await _referralService.SendEmployerReferralEmail(dto.OpportunityId);
+            await _referralService.SendProviderReferralEmail(dto.OpportunityId);
```
Comment (adamesfa) [inline src/Sfa.Tl.Matching.Web/Controllers/OpportunityController.cs]: We could probably do these in parallel using Task.WhenAll as they aren't dependant on eachother

File: src/Sfa.Tl.Matching.Application/Services/ReferralService.cs (hunk 1 above)
Comment (mikeymonster) [reply src/Sfa.Tl.Matching.Application/Services/ReferralService.cs]: You're the second person to say that, but I see it as a method with parameters, not something that is passing data around. Can consider as a future refactoring.

File: src/Sfa.Tl.Matching.Web/Controllers/OpportunityController.cs (hunk 3 above)
Comment (mikeymonster) [reply src/Sfa.Tl.Matching.Web/Controllers/OpportunityController.cs]: Changed as suggested.

**Pull request merged**